text/0026-rest-api/rest-api.yaml

```diff
@@ -0,0 +1,983 @@
+            latestMilestoneIndex:
+              type: integer
+            solidMilestoneMessageId:
+              type: string
+            solidMilestoneIndex:
+              type: integer
+            pruningIndex:
+              type: integer
+            features:
+              type: array
+              items:
+                type: string
+  
+    'GetTipsResponse':
+      properties:
+        data:
+          type: object    
+          properties:
+            tip1:
+              type: string
+            tip2:
+              type: string
+
+    'PostMessagesResponse':
+      properties:
+        data:    
+          type: object    
+          properties:        
+            messageId:
+              type: string
+
+    'GetMessagesByIndexResponse':
+      properties:
+        data:    
+          type: object    
+          properties:  
+            index:
+             type: string
+            maxResults:
+             type: integer
+            count:
+             type: integer
+            messageIds:
+             type: array
+             items:
+               type: string
+
+    'GetMessageMetadataByIdResponse':
+      properties:
+        data:
+          type: object    
+          properties:
+            messageId:
+              type: string
+            parent1MessageId:
+              type: string
+            parent2MessageId:
+              type: string
+            solid:
+              type: boolean
+            referencedByMilestone:
+              type: integer
+              nullable: true
+            ledgerInclusionState:
+              enum: 
+                - included
+                - conflicting
+            shouldPromote:
+              type: boolean
+            shouldReattach:
+              type: boolean
+          required:
+            - messageId
+            - parent1MessageId
+            - parent2MessageId
+            - solid
+            - referencedByMilestone
+
+    'GetMessageByIdDataResponse':
+      properties:
+        data:
+          type: object
+          properties:
+            allOf:
+              $ref: '#/components/schemas/Message'
+  
+    'GetMessageChildrenResponse':
+      properties:
+        data:
+          type: object    
+          properties:
+            messageId:
+              type: string
+            maxResults:
+              type: integer
+            count:
+              type: integer
+            children:
+              type: array
+              items:
+                type: string
+
+    'GetOutputsByIdResponse':
+      properties:
+        data:
+          type: object
+          properties:
+            messageId:
+              type: string
+            transactionId:
+              type: string
+            outputIndex:
+              type: integer
+            spent:
+              type: boolean
+            output:
+              anyOf:
+                - $ref: '#/components/schemas/SigLockedSingleOutput'
+
+    'GetAddressBalanceResponse':
```

Comment:
Could it also has a field telling if this address is spent?

Reply:
A reusable address cannot be "spent". You either have outputs or not. The single outputs can be spent. To get the outputs use the other endpoint. This is just an "easy" way to get the balance of the address.

Reply:
Got it! yeah it's nice to see there's a way to get address balance without several endpoint calls.